packages/date-picker/src/DatePicker/DatePickerInput/DatePickerInput.tsx

```diff
@@ -108,40 +109,38 @@ export const DatePickerInput = forwardRef<HTMLDivElement, DatePickerInputProps>(
 
       const isSegmentEmpty = !target.value;
 
-      const { selectionStart, selectionEnd } = target;
-
       switch (key) {
         case keyMap.ArrowLeft: {
+          // Prevent the cursor on tab and initial click
+          e.preventDefault();
```

Comment:
Why would `e.preventDefault` in a keyboard handler prevent the cursor on a tab/click event?

Reply:
I should reword this; without this, the input does not select all the text.